### PR TITLE
[NUI][AT-SPI] Simplify AccessibilityStates

### PR DESCRIPTION
**Pull request by grzelewski**
Replace class AccessibilityStates with enum flag.

Requires:
https://review.tizen.org/gerrit/#/c/platform/core/uifw/dali-csharp-binder/+/256381/
https://review.tizen.org/gerrit/#/c/platform/core/uifw/dali-csharp-binder/+/257412/

### Description of Change ###
Simplify usage of AccessibilityStates. This change also reduce number of interops required to support States.

### API Changes ###
None

Changed:
- Replaced AccessibilityStates class with enum flag

### CODE REVIEW

**Review by lol-github — APPROVED** [review]
LGTM, verified successfully in nui-test-app, last unmerged dependency should be merged first:
https://review.tizen.org/gerrit/#/c/platform/core/uifw/dali-csharp-binder/+/256381/

**Comment by Seoyeon2Kim** [comment]
All patches for dali-csharp-binder are merged now.
So, after checking them, please update the title of this PR :)

**Comment by lol-github** [comment]
> All patches for dali-csharp-binder are merged now.
> So, after checking them, please update the title of this PR :)

PR has been rebased and verified successfully so it is ready to merge. Unfortunately during the absence of author by 15th May we are not able to edit the PR title. In case you have such ability please feel free to update the title.

**Review by lol-github — APPROVED** [review]
LGTM

File: src/Tizen.NUI/src/public/BaseComponents/ViewAccessibilityEnum.cs [hunk 1]
```diff
@@ -146,6 +146,21 @@ public enum AccessibilityState
     /// <since_tizen> 3 </since_tizen>
     public partial class View
     {
+        [EditorBrowsable(EditorBrowsableState.Never)]
```
Comment (Seoyeon2Kim) [inline src/Tizen.NUI/src/public/BaseComponents/ViewAccessibilityEnum.cs:149]: Please add a summary for this method with the role of parameters.

File: src/Tizen.NUI/src/public/BaseComponents/ViewAccessibilityEnum.cs [hunk 2]
```diff
@@ -89,55 +89,55 @@ public enum AccessibilityGestureState
 
 
     [EditorBrowsable(EditorBrowsableState.Never)]
-    public enum AccessibilityState
+    [Flags]
+    public enum AccessibilityStates : ulong
```
[the remark this answers is not in the document]
Comment (lol-github) [reply src/Tizen.NUI/src/public/BaseComponents/ViewAccessibilityEnum.cs:93]: Currently ATSPI supports 45 different states (and this number may raise in the future) so System.Int32 type wouldn't have sufficient capacity. We spent 1 bit do denote 1 state in order to ensure simple production of state combinations with bit operators.

File: src/Tizen.NUI/src/public/BaseComponents/ViewAccessibilityEnum.cs (hunk 1 above)
Comment (lol-github) [reply src/Tizen.NUI/src/public/BaseComponents/ViewAccessibilityEnum.cs:149]: I propose change of "v" parameter to "state" and addition of the following description :

        /// <summary>
        /// A helper method to manipulate individual bit flags (e.g. turn them on or off)
        /// </summary>
        /// <param name="obj">An object that accumulates combination of bit flags</param>
        /// <param name="bit">A bit flag to be operated</param>
        /// <param name="state">A state of the bit flag to be set (0 == off, 1 ==on)</param>

**Review by lol-github — CHANGES_REQUESTED** [review]

**Review by lol-github — APPROVED** [review]
LGTM

**Review by lol-github — APPROVED** [review]
LGTM, verified successfully in nui-test-app

**Review by lol-github — APPROVED** [review]
LGTM, verified successfully after rebase

**Review by dongsug-song — APPROVED** [review]
looks good to me

**Pull request merged**